experimental/QuadFormAndIsom/src/spaces_with_isometry.jl

```diff
@@ -0,0 +1,796 @@
+
+###############################################################################
+#
+#  Accesors
+#
+###############################################################################
+
+@doc raw"""
+    space(Vf::QuadSpaceWithIsom) -> QuadSpace
+
+Given a quadratic space with isometry $(V, f)$, return the underlying space `V`.
+
+# Examples
+```jldoctest
+julia> V = quadratic_space(QQ, 2);
+
+julia> Vf = quadratic_space_with_isometry(V; neg = true);
+
+julia> space(Vf) === V
+true
+```
+"""
+space(Vf::QuadSpaceWithIsom) = Vf.V
+
+@doc raw"""
+    isometry(Vf::QuadSpaceWithIsom) -> QQMatrix
+
+Given a quadratic space with isometry $(V, f)$, return the underlying isometry
+`f`.
+
+# Examples
+```jldoctest
+julia> V = quadratic_space(QQ, 2);
+
+julia> Vf = quadratic_space_with_isometry(V; neg = true);
+
+julia> isometry(Vf)
+[-1    0]
+[ 0   -1]
+```
+"""
+isometry(Vf::QuadSpaceWithIsom) = Vf.f
+
+@doc raw"""
+    order_of_isometry(Vf::QuadSpaceWithIsom) -> IntExt
+
+Given a quadratic space with isometry $(V, f)$, return the order of the
+underlying isometry `f`.
+
+# Examples
+```jldoctest
+julia> V = quadratic_space(QQ, 2);
+
+julia> Vf = quadratic_space_with_isometry(V; neg = true);
+
+julia> order_of_isometry(Vf) == 2
+true
+```
+"""
+order_of_isometry(Vf::QuadSpaceWithIsom) = Vf.n
+
+###############################################################################
+#
+#  Attributes
+#
+###############################################################################
+
+@doc raw"""
+    rank(Vf::QuadSpaceWithIsom) -> Integer
+
+Given a quadratic space with isometry $(V, f)$, return the rank of the underlying
+space `V`.
+
+# Examples
+```jldoctest
+julia> V = quadratic_space(QQ, 2);
+
+julia> Vf = quadratic_space_with_isometry(V; neg = true);
+
+julia> rank(Vf) == 2
+true
+```
+"""
+rank(Vf::QuadSpaceWithIsom) = rank(space(Vf))::Integer
```

Comment:
```suggestion
rank(Vf::QuadSpaceWithIsom) = rank(space(Vf))
```
Should not be necessary?